### DISCUSSION TITLE
Auto delete genres

### DISCUSSION THREAD

**Discussion by tarz4n** [beetbox/beets#5429] [Q&A]
Hello,

Been setting up Beets and love it! One thing i have not figured out how to do.

Im importing alot of music, and some genres i just dont want in my collection. Are there any way to just auto delete anything that have “RNB” in the genre, while im automatic importing?

**Comment by JOJ0** [comment]
There is several ways.

Something automatic but depending on an external service is the `lastgenre` plugin. It's highly configurable, but not perfect yet. Also consider this open PR #4982, it adds some more flexibility but will only finish in a month or so...I hope ;-) 

If you don't want to rely on what last.fm gives you as genres or for now only fix "major problems" manually, `beet modify -a genre="Rhythm & Blues" genre:rnb` might help.

This only works for "exact matches" of `genre:rnb` and applies to albums (remove `-a` to match tracks). If you want to "search and replace" within the genre tag, that unfortunately currently is not possible. There was an open feature request for that somewhere......

**Comment by tarz4n** [comment]
Thank you for your reply, @JOJ0 

I dont use lastgenre, as i am relying on the genre in the files themselves. I trying to figure out how Beets can automatically delete certain files while importing. I dont want to modify, just delete them instead of importing.

**Reply by JOJ0** [reply]
Sorry I misunderstood. I thought you want to alter the genre tag during importing....

Deleting is not possible during import. You can skip what you don't want and later on go through in the log file and delete all paths manually

config log file docs here https://beets.readthedocs.io/en/latest/reference/config.html#log

**Reply by JOJ0** [reply]
If you feel adventurous and have a solid backup strategy for your music, you can install beets from git and try this new plugin https://github.com/beetbox/beets/pull/4748

it has the feature you want partly: you can decide to delete original source paths during import

git install docs here: https://beets.readthedocs.io/en/latest/contributing.html#getting-the-source

**Comment by arsaboo** [comment]
Another approach would be to move `RNB` genre to a different folder and then delete the folder. This may be easier as you will have the chance to review the files before you delete them. Here's a snippet from my config that defines the folder based on genre (feel free to modify this based on your setting):

```yaml
paths:
    default: /data/music/$folder/$initial_char/$album%aunique{album year,,[]} ($year)/$album ($year) - $title
    singleton: /data/music/$folder/$initial_char/$album%aunique{album year,,[]} ($year)/$album ($year) - $title
    comp: /data/music/$folder/$initial_char/$album%aunique{album year,,[]} ($year)/$album ($year) - $title
    albumtype_soundtrack: /data/music/$folder/$initial_char/$album%aunique{album year,,[]} ($year)/$album ($year) - $title

item_fields:
   initial_char: |
       char = album[0].capitalize()
       if album[0].isnumeric():
           char = '#'
       return char
   folder: |
       folder = 'XYZ'
       if genre.lower()=='rnb':
           folder = 'RNB'
       elif genre.lower()=='classical':
           folder = 'Classical Sorted'
       return folder
```

**Comment by tarz4n** [comment]
@arsaboo,

Thank you. So, if i want to delete everything i import with **House** and  **Classical**  in the genre and want it to be put in /(Library)/INSPECT/House/ or /(Library)/INSPECT/Classical Sorted/

```
item_fields:
   initial_char: |
       char = album[0].capitalize()
       if album[0].isnumeric():
           char = '#'
       return char
   folder: |
       folder = 'INSPECT'
       if genre.lower()=='house':
           folder = 'House'
       elif genre.lower()=='classical':
           folder = 'Classical Sorted'
       return folder
```

I tried it out, and it ignores it and sort like i have it in my path:

```
plugins: fetchart embedart convert scrub replaygain web zero advancedrewrite inline
paths:
    singleton: $genre/$artist - $title ($year)
```

**Comment by arsaboo** [comment]
Because you are not using `folder` in the path. You must modify the `paths` to include `folder`.

**Reply by JOJ0** [reply]
$folder in the path, yeah that's missing. 

Also make sure you have the "inline" plugin active!

https://beets.readthedocs.io/en/latest/plugins/inline.html
